### PR TITLE
User defaults group migration

### PR DESCRIPTION
**Pull request by llbartekll**
# Description

Resolves # (issue)

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Sources/Auth/AuthClientFactory.swift [hunk 1]
```diff
@@ -6,10 +6,11 @@ public struct AuthClientFactory {
         projectId: String,
         crypto: CryptoProvider,
         networkingClient: NetworkingInteractor,
-        pairingRegisterer: PairingRegisterer
+        pairingRegisterer: PairingRegisterer,
+        groupIdentifier: String
     ) -> AuthClient {
         let logger = ConsoleLogger(loggingLevel: .off)
-        let keyValueStorage = UserDefaults.standard
+        let keyValueStorage = UserDefaults(suiteName: groupIdentifier) ?? UserDefaults.standard
```
Comment (llbartekll) [inline Sources/Auth/AuthClientFactory.swift:13]: @flypaper0 what do you think of fallback to UserDefaults .standard?

Comment (flypaper0) [reply Sources/Auth/AuthClientFactory.swift:13]: with fallback we're give developers a chance to make mistake. I'd prefer fatal error here

Comment (flypaper0) [reply Sources/Auth/AuthClientFactory.swift:13]: check NotifyClientFactory line 104

File: Sources/Auth/Auth.swift [hunk 2]
```diff
@@ -28,7 +28,8 @@ public class Auth {
             projectId: Networking.projectId,
             crypto: config.crypto,
             networkingClient: Networking.interactor,
-            pairingRegisterer: Pair.registerer
+            pairingRegisterer: Pair.registerer,
+            groupIdentifier: Networking.groupIdentifier
```
Comment (flypaper0) [inline Sources/Auth/Auth.swift:32]: cool, Notify also can get groupIdentifier from Networking than

File: Sources/WalletConnectNetworking/Networking.swift [hunk 3]
```diff
@@ -8,13 +8,20 @@ public class Networking {
     }
 
     public static var interactor: NetworkingInteractor = {
-        guard let _ = Networking.config else {
```
Comment (flypaper0) [inline Sources/WalletConnectNetworking/Networking.swift:11]: It was empty? lol

File: Sources/WalletConnectNetworking/Networking.swift [hunk 4]
```diff
@@ -8,13 +8,20 @@ public class Networking {
...
+        guard let config = Networking.config else {
             fatalError("Error - you must call Networking.configure(_:) before accessing the shared instance.")
         }
 
-        return NetworkingClientFactory.create(relayClient: Relay.instance)
+        return NetworkingClientFactory.create(relayClient: Relay.instance, groupIdentifier: config.groupIdentifier)
     }()
 
+    public static var groupIdentifier: String {
+        guard let config = Networking.config else {
```
Comment (flypaper0) [inline Sources/WalletConnectNetworking/Networking.swift:19]: we're copy and pasting fatal error. May be create `private static var config: Config` and throw fatal error only there?

File: Sources/WalletConnectNetworking/NetworkingClientFactory.swift [hunk 5]
```diff
@@ -2,9 +2,12 @@ import Foundation
...
 public struct NetworkingClientFactory {
 
-    public static func create(relayClient: RelayClient) -> NetworkingInteractor {
+    public static func create(
+        relayClient: RelayClient,
+        groupIdentifier: String
+    ) -> NetworkingInteractor {
         let logger = ConsoleLogger(prefix: "🕸️", loggingLevel: .off)
-        let keyValueStorage = UserDefaults.standard
+        let keyValueStorage = UserDefaults(suiteName: groupIdentifier) ?? UserDefaults.standard
```
Comment (flypaper0) [inline Sources/WalletConnectNetworking/NetworkingClientFactory.swift:10]: still not sure about fallback.. let's discuss on sync

File: Sources/Auth/Auth.swift (hunk 2 above)
Comment (flypaper0) [reply Sources/Auth/Auth.swift:32]: nvm, I see you done it 👍

File: Sources/WalletConnectPush/PushClientFactory.swift [hunk 6]
```diff
@@ -1,12 +1,15 @@
...
+    public static func create(
+        projectId: String,
+        pushHost: String,
+        groupIdentifier: String,
+        environment: APNSEnvironment
+    ) -> PushClient {
 
         let keychainStorage = KeychainStorage(serviceIdentifier: "com.walletconnect.sdk")
-        let keyValueStorage = UserDefaults.standard
+        let keyValueStorage = UserDefaults(suiteName: groupIdentifier) ?? UserDefaults.standard
```
Comment (flypaper0) [inline Sources/WalletConnectPush/PushClientFactory.swift:12]: let's consider Factory?

File: Sources/WalletConnectRelay/ClientAuth/ClientIdStorage.swift [hunk 7]
```diff
@@ -50,20 +51,41 @@ private extension ClientIdStorage {
...
             setPublicPart(privateKey.publicKey)
             try keychain.delete(key: oldStorageKey)
-            logger.debug("ClientID migrated")
+            logger.debug("Keychain data migrated to group UserDefaults")
         } catch {
             logger.debug("ClientID migration failed with: \(error.localizedDescription)")
         }
     }
 
+    private func migratePublicKeyToGroupUserDefaultsIfNeeded() {
```
Comment (flypaper0) [inline Sources/WalletConnectRelay/ClientAuth/ClientIdStorage.swift:74]: Why we're migrating it separately? 
I meant something like: 
```
for key, value in UserDefaults.standard.dictionaryRepresentation() {
  UserDefaults(groupIdentifier).set(value, for: key)
}
```

**Pull request merged**